feat: workspaces dashboard MVP

nickytonline opened the pull request:
## Description

This is in a good state for the demo for @bdougie/MVP. Follow up PRs will fill in the gaps.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Relates to #2438

## Mobile & Desktop Screenshots/Recordings

![CleanShot 2024-01-30 at 16 56 55](https://github.com/open-sauced/app/assets/833231/e3d07b5f-72e0-4bd7-9fe0-972ec15765b7)

https://github.com/open-sauced/app/assets/833231/c70d85bb-bfeb-41ae-af56-3787cab9004d

## Steps to QA

1. Ensure that you're logged in.
2. Navigate to `/workspaces/new` (we currently don't have landing pages enabled yet)
3. Create a new workspace, or load an existing one up from the app sidebar top select menu (Ensure there is at least one repository that the workspace is tracking).
4. The workspace dashboard loads.
5. Use the day range picker to change the range between 7, 30, and 90 days. Notice the grid refreshes with information for that particular day range.

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?



## [optional] What gif best describes this PR or how it makes you feel?

<img src="https://media0.giphy.com/media/biYxmbBKOTGk6TQJ1G/giphy.gif"/>

<!-- note: PRs with deleted sections will be marked invalid -->

<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
  
  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.
  
  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Open Sauced Contributing Guide: https://github.com/open-sauced/.github/blob/main/CONTRIBUTING.md.
  - 📖 Read the Open Sauced Code of Conduct: https://github.com/open-sauced/.github/blob/main/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

jpmcb commented:
How can I actually get to the workspace dashboard? After creating it from `workspaces/new`, I update it and just get the settings page:

![Screenshot 2024-01-30 at 3 03 55 PM](https://github.com/open-sauced/app/assets/23109390/32034dcd-33cd-4ccd-a7c7-f89e6b7489b8)

nickytonline commented on components/Workspaces/RepositoryStatCard.stories.tsx:
```diff
@@ -20,6 +21,13 @@ export const PullRequests: Story = {
   },
 };
 
+export const Error: Story = {
```
Adds an error state Storybook story for the repo stats cards.

![CleanShot 2024-01-30 at 17 06 23](https://github.com/open-sauced/app/assets/833231/399505e5-918d-4c53-a65f-11b98b67cd45)

nickytonline commented on components/Workspaces/RepositoryStatCard.tsx:
```diff
@@ -91,11 +103,13 @@ export const RepositoryStatCard = ({ stats, type, isLoading }: RepositoryStatCar
                 ) : (
-                  <td className="semi-bold text-2xl">{stat === "velocity" ? `${Math.round(value)}d` : value}</td>
+                  <td className="semi-bold text-2xl" title={`${value}`}>
+                    {stat === "velocity" ? `${Math.round(value)}d` : humanizeNumber(value, "abbreviation")}
```
We're humanizing the numbers here because in some cases, like Kubernetes, the numbers are huge.

![image](https://github.com/open-sauced/app/assets/833231/b4e071b9-5bfb-43de-bfea-43883a34ad81)

nickytonline commented on components/Workspaces/WorkspacesTabList.tsx:
```diff
@@ -0,0 +1,25 @@
+];
+
+interface WorkspacesTabListProps {
+  selectedTab: TabName;
```
I narrowed the type so that the `selectedTab` has to be one of the tab names.

nickytonline commented on components/organisms/Repositories/repositories.tsx:
```diff
@@ -20,9 +20,10 @@ import RepoNotIndexed from "./repository-not-indexed";
 }
 
-const Repositories = ({ repositories }: RepositoriesProps): JSX.Element => {
+const Repositories = ({ repositories, showSearch = true }: RepositoriesProps): JSX.Element => {
```
In the case of the workspaces dashboard, we do not want this search enabled.

nickytonline commented on components/shared/AppSidebar/AppSidebar.tsx:
```diff
@@ -73,16 +75,15 @@ export const AppSideBar = () => {
 
-              setWorkspaceId(value);
-              window.location.href = `/workspaces/${value}/settings`;
+              window.location.href = `/workspaces/${value}/repositories`;
```
Now when you select a workspace from the app sidebar, it loads the dashboard instead of the settings page.

nickytonline commented on pages/workspaces/new.tsx:
```diff
@@ -47,7 +47,7 @@ const NewWorkspace = () => {
 
   return (
-    <WorkspaceLayout>
+    <WorkspaceLayout workspaceId={null}>
```
Since the new workspace page won't have a workspace ID, we set this to `null`. I didn't make it an optional field as it really is required. This is just a special case, hence the `null`.

nickytonline commented:
> How can I actually get to the workspace dashboard? After creating it from `workspaces/new`, I update it and just get the settings page:
> 
> ![Screenshot 2024-01-30 at 3 03 55 PM](https://private-user-images.githubusercontent.com/23109390/300974275-32034dcd-33cd-4ccd-a7c7-f89e6b7489b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2NTI5OTQsIm5iZiI6MTcwNjY1MjY5NCwicGF0aCI6Ii8yMzEwOTM5MC8zMDA5NzQyNzUtMzIwMzRkY2QtMzNjZC00Y2NkLWE3YzctZjg5ZTZiNzQ4OWI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMwVDIyMTEzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5MGM2NmZlYWNkZWE4NDIzYjljODZjYzY1MWI2YzY3N2UxMzJhMTk5ZmIxOGNjZTlkMGNmODZkMzJiODM5YTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5a9hTVA-TSPtIZHQ34iz3i8vyM2fEMtl0x0l5J8RnDs)
> 
> Dropping `settings` from the URL also just takes me to this weird "Workspaces" topic page:
> 
> ![Screenshot 2024-01-30 at 3 05 19 PM](https://private-user-images.githubusercontent.com/23109390/300974701-06b7b876-44a9-4bbf-9cc5-6df64e000beb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2NTI5OTQsIm5iZiI6MTcwNjY1MjY5NCwicGF0aCI6Ii8yMzEwOTM5MC8zMDA5NzQ3MDEtMDZiN2I4NzYtNDRhOS00YmJmLTljYzUtNmRmNjRlMDAwYmViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMwVDIyMTEzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4NzJiNGI3ZTBhNzNiNzZhNTk0YWNlMmQ2YTcwZjY5YjJiYzgyMzUzOThjMzkwMmMwOTkyMzQ0YzY5ZjlmYzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.csTl49ESXfAKz2gbEEpEC2BC3SL9BFQAy92_K8lUhqI)

Good catch @jpmcb. If you select an existing workspace from the drop down it goes to your dashboard, but not after creating one. I'll fix that.

nickytonline commented:
@isabensusan, I added an empty state to the dashboard if they didn't add any repositories to track. It's basically a CTA that takes them back to the settings page. I also added the tabs to the settings page so they can navigate back to the dashboard.

![CleanShot 2024-01-30 at 18 20 34](https://github.com/open-sauced/app/assets/833231/7f403986-266c-45e0-9776-e210a2353448)

Let me know if this is OK or if you'd like it tweaked. It came about after @jpmcb's [comment](https://github.com/open-sauced/app/pull/2554#issuecomment-1917973345).

jpmcb reviewed (approved):
Very solid Nick - great work!!! 👏🏼 🕺🏼 

---

At some point, probably not in this MVP, we'll want to think about responsiveness of this dashboard and the whole workspaces flow:

![Screenshot 2024-01-30 at 4 27 26 PM](https://github.com/open-sauced/app/assets/23109390/c7f323e3-43d0-40f5-9052-83de54511c32)

This gets _abit_ squished if I have my window smaller than this in my ultrawide monitor.

nickytonline commented:
> Very solid Nick - great work!!! 👏🏼 🕺🏼 
> 
> 
> 
> ---
> 
> 
> 
> At some point, probably not in this MVP, we'll want to think about responsiveness of this dashboard and the whole workspaces flow:
> 
> 
> 
> ![Screenshot 2024-01-30 at 4 27 26 PM](https://github.com/open-sauced/app/assets/23109390/c7f323e3-43d0-40f5-9052-83de54511c32)
> 
> 
> 
> This gets _abit_ squished if I have my window smaller than this in my ultrawide monitor.

Yeah, we have new responsive tables I introduced as part of workspaces, but this is an existing table component that needs some love. On the TODO for the app in general.

bdougie commented:
> How can I actually get to the workspace dashboard? After creating it from `workspaces/new`, I update it and just get the settings page:
> 
> ![Screenshot 2024-01-30 at 3 03 55 PM](https://private-user-images.githubusercontent.com/23109390/300974275-32034dcd-33cd-4ccd-a7c7-f89e6b7489b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2NTk1MDksIm5iZiI6MTcwNjY1OTIwOSwicGF0aCI6Ii8yMzEwOTM5MC8zMDA5NzQyNzUtMzIwMzRkY2QtMzNjZC00Y2NkLWE3YzctZjg5ZTZiNzQ4OWI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDAwMDAwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxOWI2ODQyNzUwZWY2NWZmZjgxOTM1OWFkMmUzMTQ1ZjAzMjNiYjI0ZWJlYTg0NmZhOTM5NTcyY2IwM2ZlYzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qG7vzrHk8x5uggY-k4QAs7MzNE5opCQufs_O_PSy1ys)
> 
> Dropping `settings` from the URL also just takes me to this weird "Workspaces" topic page:

You can add /repositories after the ID. `workspaces/123456/repositories`

I also get there by choosing my workspace from the dropdown. @nickytonline covered this navigation in our meeting as a workaround, but plans to cover that in a follow up PR/commit.

brandonroberts reviewed (approved):
🚢
